src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -64,6 +65,118 @@ export default class ReimbursementRequestService {
+   * @param sender the person sending the pending advisor list
+   * @param saboNumbers the sabo numbers of the reimbursement requests to send
+   */
+  static async sendPendingAdvisorList(
+    sender: User & {
+      teams: Team[];
+    },
+    saboNumbers: number[]
+  ) {
+    const financeTeam = await prisma.team.findUnique({
+      where: { teamId: process.env.FINANCE_TEAM_ID }
+    });
+
+    if (!financeTeam) throw new HttpException(500, 'Finance team does not exist!');
+
+    if (
+      !sender.teams.some((team) => team.teamId === process.env.FINANCE_TEAM_ID) &&
+      !(financeTeam.leaderId === sender.userId)
+    ) {
+      throw new AccessDeniedException(`You are not a member of the finance team!`);
+    }
+
+    const reimbursementRequests = await prisma.reimbursement_Request.findMany({
+      where: {
+        saboId: {
+          in: saboNumbers
+        }
+      }
+    });
+
+    if (reimbursementRequests.length < saboNumbers.length) {
+      const saboNumbersNotFound = saboNumbers.filter((saboNumber) => {
+        return !reimbursementRequests.some((reimbursementRequest) => reimbursementRequest.saboId === saboNumber);
+      });
+      throw new HttpException(400, `The following sabo numbers do not exist: ${saboNumbersNotFound.join(', ')}`);
+    }
+
+    const deletedReimbursementRequests = reimbursementRequests.filter(
+      (reimbursementRequest) => reimbursementRequest.dateDeleted
+    );
+
+    if (deletedReimbursementRequests.length > 0) {
+      const saboNumbersDeleted = deletedReimbursementRequests.map((reimbursementRequest) => reimbursementRequest.saboId);
+      throw new HttpException(
+        400,
+        `The following reimbursement requests with these sabo numbers have been deleted: ${saboNumbersDeleted.join(', ')}`
+      );
+    }
+
+    const mailOptions = {
+      subject: 'Reimbursement Requests To Be Approved By Advisor',
+      text: `The following reimbursement requests need to be approved by you: ${saboNumbers.join(', ')}`
+    };
+
+    await sendEmail(mailOptions.subject, mailOptions.text);
+
+    reimbursementRequests.forEach((reimbursementRequest) => {
+      prisma.reimbursement_Status.create({
+        data: {
+          type: Reimbursement_Status_Type.ADVISOR_APPROVED,
+          userId: sender.userId,
+          reimbursementRequestId: reimbursementRequest.reimbursementRequestId
+        }
+      });
+    });
+  }
+
+  static async addSaboNumber(reimbursementRequestId: string, saboNumber: number, submitter: User & { teams: Team[] }) {
+    const financeTeam = await prisma.team.findUnique({
+      where: { teamId: process.env.FINANCE_TEAM_ID }
+    });
+
+    if (!financeTeam) throw new HttpException(500, 'Finance team does not exist!');
+
+    if (
+      !submitter.teams.some((team) => team.teamId === process.env.FINANCE_TEAM_ID) &&
+      !(financeTeam.leaderId === submitter.userId)
+    ) {
+      throw new AccessDeniedException(`You are not a member of the finance team!`);
+    }
```

Comment:
could you abstract lines 138 - 149 into a helper method since you're using it in both service functions